internal/newusers/manager_test.go

```diff
@@ -0,0 +1,577 @@
+		"Do not prevent cache creation if cleanup fails":         {dbFile: "multiple_users_and_groups", procDir: "does-not-exist"},
+		"Do not stop cache if cleanup routine fails":             {dbFile: "multiple_users_and_groups", procDir: "does-not-exist", skipCleanOnNew: true, cleanupInterval: 1},
+		"Do not clean user if can not get groups":                {dbFile: "invalid_entry_in_userToGroups", expirationDate: "2020-01-01"},
+		"Do not clean user if can not delete user from group":    {dbFile: "invalid_entry_in_groupByID", expirationDate: "2020-01-01"},
+
+		// Corrupted databases
+		"New recreates any missing buckets and delete unknowns": {dbFile: "database_with_unknown_bucket"},
+		"Database flagged as dirty is cleared up":               {dbFile: "multiple_users_and_groups", dirtyFlag: true},
+		"Corrupted database when opening is cleared up":         {corruptedDbFile: true},
+		"Dynamically mark database as corrupted is cleared up":  {markDirty: true},
+
+		"Error if cacheDir does not exist": {dbFile: "-", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			cacheDir := t.TempDir()
+			if tc.dbFile == "-" {
+				err := os.RemoveAll(cacheDir)
+				require.NoError(t, err, "Setup: could not remove temporary cache directory")
+			} else if tc.dbFile != "" {
+				createDBFile(t, filepath.Join("testdata", tc.dbFile+".db.yaml"), cacheDir)
+			}
+			if tc.dirtyFlag {
+				err := os.WriteFile(filepath.Join(cacheDir, cachetests.DirtyFlagDbName), nil, 0600)
+				require.NoError(t, err, "Setup: could not create dirty flag file")
+			}
+			if tc.corruptedDbFile {
+				err := os.WriteFile(filepath.Join(cacheDir, cachetests.DbName), []byte("Corrupted db"), 0600)
+				require.NoError(t, err, "Setup: Can't update the file with invalid db content")
+			}
+
+			if tc.expirationDate == "" {
+				tc.expirationDate = "2004-01-01"
+			}
+			expiration, err := time.Parse(time.DateOnly, tc.expirationDate)
+			require.NoError(t, err, "Setup: could not calculate expiration date for tests")
+			managerOpts := []newusers.Option{newusers.WithUserExpirationDate(expiration)}
+
+			if tc.cleanupInterval > 0 {
+				managerOpts = append(managerOpts, newusers.WithCacheCleanupInterval(time.Second*time.Duration(tc.cleanupInterval)))
+			}
+			if tc.skipCleanOnNew {
+				managerOpts = append(managerOpts, newusers.WithoutCleaningCacheOnNew())
+			}
+			if tc.procDir != "" {
+				managerOpts = append(managerOpts, newusers.WithProcDir(tc.procDir))
+			}
+
+			m, err := newusers.NewManager(cacheDir, managerOpts...)
+			if tc.wantErr {
+				require.Error(t, err, "NewManager should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "NewManager should not return an error, but did")
+			t.Cleanup(func() { _ = m.Stop() })
+
+			// Wait for the cleanup routine
+			time.Sleep(3 * time.Second)
```

Comment:
You know, I'm not a fan of sleeping in tests, really we should have something that waits until a condition is met or fail otherwise.

Reply:
I know, but this is a routine that we don't have "control" of. We don't want nor need to sync on its execution. This is mainly to see that it's working.